Added a Volunteer Permission Level

sarj21 opened the pull request:
We don't want volunteers being able to edit people or see stats or send mass emails so this should allow them to only see users and check them in

sarj21 committed:
Added a Volunteer Permission Level

mecolmg commented:
We should also hide the checkbox and the link button (see below)
![image](https://user-images.githubusercontent.com/8546611/35692501-854e237e-0749-11e8-8ec2-c37af9acbf80.png)

sarj21 commented:
Oh I should hide the checkbox and also check that it bounces volunteers. I think I want to leave the link button so volunteers can pull up a persons details if they need to. I might just want to scrub the results of some stuff they dont need to see. We can ask the others what they think

sarj21 committed:
Hide checkboxes and block email section

now they can't see the checkboxes next to peoples names and they cant get to the mass email page

sarj21 commented:
This seems to work now. The checkboxes are hidden from volunteers and they can't hit any important admin only pages. Also, I stepped them down to demi-gods

mecolmg commented on app/client/views/admin/users/users.html:
```diff
@@ -97,7 +97,7 @@ <h3> {{section.title}} </h3>
         <thead>
           <tr>
-            <th></th>
+            <th ng-if="selectedUser.admin"></th>
```
Should be `currentUser.admin`

This currently breaks admin view normally.

mecolmg reviewed (changes requested):
Also, the volunteers currently don't have access to the `user` page that the little link button goes to, mainly because that page allows updating user info. Volunteers can see user info by clicking on the row, so there's no reason for them to see the link.

sarj21 committed:
Fixed volunteer source user

sarj21 committed:
Hid edit link for users from volunteers

It will also bounce any requests on the api so just hiding it is good enough

sarj21 commented:
Yeah my confusion, I forgot what the link button did. And I used the wrong user source. It looks like its fixed
